specification.markdown

```diff
@@ -64,30 +64,67 @@ followed by the Data Size, and then the non-interpreted Binary itself:
-        0000 01xx  xxxx xxxx  xxxx xxxx  xxxx xxxx  xxxx xxxx  xxxx xxxx                       - value 0 to 2^42-2
-        0000 001x  xxxx xxxx  xxxx xxxx  xxxx xxxx  xxxx xxxx  xxxx xxxx  xxxx xxxx            - value 0 to 2^49-2
-        0000 0001  xxxx xxxx  xxxx xxxx  xxxx xxxx  xxxx xxxx  xxxx xxxx  xxxx xxxx  xxxx xxxx - value 0 to 2^56-2
-
-    Since modern computers do not easily deal with data coded in sizes
-    greater than 64 bits, any larger Element Sizes are left undefined at
-    the moment. Currently, the Element Size coding allows for an Element
-    to grow to 72000 To, i.e. 7x10\^16 octets or 72000 terabytes, which
-    will be sufficient for the time being.
-
-    There is only one reserved word for Element Size encoding, which is
-    an Element Size encoded to all 1's. Such a coding indicates that the
-    size of the Element is unknown, which is a special case that we
-    believe will be useful for live streaming purposes. However, avoid
-    using this reserved word unnecessarily, because it makes parsing
-    slower and more difficult to implement.
+## Element Data Size
+
+The Element Data Size expresses the length in octets of Element Data.
+The Element Data Size itself MUST be encoded as a Variable Size Integer.
+By default, EBML Element Data Sizes can be encoded in lengths from one
+octet to eight octets, although Element Data Sizes of greater lengths
+MAY be used if the octet length of the EBML document's longest Element
+Data Size is declared in the EBMLMaxSizeLength Element of the EBML
+Header. Unlike the VINT\_DATA of the Element ID, the VINT\_DATA
+component of the Element Data Size is NOT REQUIRED to be encoded at the
+shortest valid length. For example, an Element Data Size with binary
+encoding of 1011 1111 or a binary encoding of 0100 0000 0011 1111 are
+both valid Element Data Sizes and both store a semantically equal value.
+
+Although an Element ID with all VINT\_DATA bits set to zero is invalid,
+an Element Data Size with all VINT\_DATA bits set to zero is allowed.
+This indicates that the Element Data of the Element is zero octets in
+length. Such an Element is referred to as an Empty Element.
+
+An Element Data Size with all VINT\_DATA bits set to one is reserved as
+an indicator that the size of the Element is unknown. This rule allows
+for an Element to be written and read before the size of the Element is
+known; however unknown Element Data Size values SHOULD NOT be used
+unnecessarily. An Element with an unknown Element Data Size MUST be a
+Master-element in that it contains other EBML Elements as sub-elements.
+The end of the Master-element is determined by the beginning of the next
+element that is not a valid sub-element of the Master-element. The use
+of Elements of unknown size is dependent on the definition of the EBML
+Schema declared in DocType, because an Element of unknown size can not
```

Comment:
Also I looked quickly in the mailing list for infinite/unknown discussions. I do not see where we reached a conclusion on limiting it. I repeatedly said there's no need for this restriction.